File: Data Structures & Algorithms/range-sum-of-bst/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        int sum = 0;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        while (!q.isEmpty()) {
            TreeNode current = q.poll();
            if (current.left != null && current.val > low) q.add(current.left);
            if (current.right != null && current.val < high) q.add(current.right);
            if (current.val >= low && current.val <= high) sum += current.val;
        }
        return sum;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        ans = []
        def dfs(cur, level):
            if not cur:
                return
            if level == len(ans):
                ans.append([])
            ans[level].append(cur.val)
            
            dfs(cur.left, level+1)
            dfs(cur.right, level + 1)
        dfs(root, 0)
        return ans
